closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1706,6 +1725,40 @@ private static void checkProviderFormat(String providerInfo, boolean update) {
                 printStackTraceAndExit("Provider format is incorrect: " + providerInfo);
             }
         }
+
+        private void loadProfileCheck(String profileID) {
+            Enumeration<?> pNames = securityProps.propertyNames();
+            String profileDot = profileID + '.';
+            while (pNames.hasMoreElements()) {
+                String name = (String) pNames.nextElement();
+                if (name.startsWith(profileDot)) {
+                    if (name.contains(".jce.provider.")) {
+                        profileCheckProviderNames.add(name);
+                    } else {
+                        profileCheckPropertyNames.add(name);
+                    }
+                }
+            }
+        }
+
+        private void checkProfileCheck(String profileID) {
+            if (!profileCheckProviderNames.isEmpty()) {
+                printStackTraceAndExit(
+                        "The order numbers of providers in profile " + profileID
+                                + " (or its base profile) are not consecutive.");
+            }
+            if (!profileCheckPropertyNames.isEmpty()) {
+                printStackTraceAndExit(
+                    "The property names: "
+                            + profileCheckPropertyNames
+                                    .stream()
+                                    .sorted()
+                                    .collect(Collectors.joining(", "))
+                            + " in profile " + profileID
+                            + " (or its base profile)"
+                            + " are not recognized.");
```

Comment:
These lines should be indented four more spaces for consistency with lines 1746-1748.

Reply:
Done.

Reply:
Profiles may be extended any number of levels, so "its" should be "a" as in " (or a base profile)".

Reply:
Done.